doc/ml_features.rst

```diff
@@ -0,0 +1,486 @@
+        model to the DB using ``Client.set_model_from_file()``. We also use
+        ``Client.put_tensor()`` to upload a synthetic MNIST sample to the DB.
+
+
+        .. code-block:: python
+
+            client = Client(cluster=False)
+            model_key = "mnist_cnn"
+            client.set_model_from_file(
+                model_key, model_path, "TF", device="GPU", inputs=inputs, outputs=outputs
+            )
+
+
+    .. group-tab:: PyTorch
+
+        PyTorch requires models to be `jit-traced <https://pytorch.org/docs/1.11/generated/torch.jit.save.html>`__.
+        The method ``torch.jit.save()`` can either store the model in memory or on file. Here,
+        we will save it to a file located at ``./traced_model.pt``.
+
+        .. code-block:: python
+
+            # Example input needed for jit tracing
+            example_forward_input = torch.rand(20, 1, 28, 28)
+            module = torch.jit.trace(n, example_forward_input)
+            model_path = "./traced_model.pt"
+            torch.jit.save(module, modelpath)
+
+
+        Now that we have the serialized model, we can upload it to the DB using
+        ``Client.set_model_from_file()`` method.
+
+        .. code-block:: python
+
+            client = Client(cluster=False)
+            model_key = "mnist_cnn"
+
+            client.set_model_from_file(model_key, model_path, "TORCH", device="CPU")
+
+
+For details about ``Client.set_model_from_file()``, please
+refer to :ref:`SmartRedis API <smartredis-api>`.
+
+===============================================
+Executing the model on tensors stored in the DB
+===============================================
+
+Now that the model is available for execution on the DB, we use the SmartRedis client
+to upload a tensor representing a batch of 20 synthetic MNIST images.
+
+.. tabs::
+
+    .. group-tab:: TensorFlow
+
+        .. code-block:: python
+
+            # 20 samples of "image" data
+            mnist_images = np.random.rand(20, 28, 28, 1).astype(np.float32)
+            # client was instantiated previously
+            client.put_tensor("mnist_images", mnist_image)
+
+
+    .. group-tab:: PyTorch
+
+        .. code-block:: python
+
+
+            # 20 samples of "image" data
+            mnist_images = torch.rand(20, 1, 28, 28)
+            # client was instantiated previously
+            client.put_tensor("mnist_images", mnist_images.numpy())
+
+
+Now we can use ``Client.run_model()`` to execute the model on the data we have
+just stored and ``Client.get_tensor()`` to donwload the output of the model execution.
+Notice that, for this part, the code is identical for models uploaded from file and from memory, and
+with TensorFlow or PyTorch backends.
+
+.. code-block:: python
+
+    client.run_model(model_key, inputs=["mnist_imagse"], outputs=["mnist_output"])
+    output = client.get_tensor("mnist_output")
+
+
+.. _ml_features_torchscript:
```

Comment:
maybe add? -> For details about ``Client.run_model()``, please
refer to :ref:`SmartRedis API <smartredis-api>`.

Reply:
Done!